packages/runtime/datastore/src/dataStoreRuntime.ts

```diff
@@ -432,19 +432,30 @@ export class FluidDataStoreRuntime
 		return context.getChannel();
 	}
 
-	public createChannel(id: string = uuid(), type: string): IChannel {
+	public createChannel(channel: IChannel): void;
```

Comment:
A method named 'createChannel(..)' than requires an IChannel as input and returns nothing is odd.

I'm guessing the input is an incomplete IChannel and calling 'createChannel(..)' completes the channel?  Is that necessary for the scenario, or do we just want the ability to provide additional arguments?

```ts
// Creates an instance of the channel with the given type.
// If provided, 'attributes' are added to the channel attributes.
createChannel({ type: string, attributes?: any }): IChannel
```

Reply:
i don't think it can work like that give how channel factories work, specifically IChannelFactory.create doesn't take attributes, created channel provides them

Reply:
the path here allows you to create a channel via any means, and then creates its context after the fact, this enables custom initialization flows.

Reply:
Yes, you can create a channel directly by its class which will also provide type checking for the attributes unlike "any" type and then just add that channel here to the runtime.